version: '3'
services:
  postgres:
    image: postgres:16
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USERNAME: ${POSTGRES_USERNAME}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    ports:
      - '5432:5432'
    volumes:
      - evalia_database:/var/lib/postgresql/data
    networks:
      - EVALIA
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d postgres" ]
      interval: 30s
      timeout: 10s
      retries: 5

  evalia:
    image: evalia
    build:
      context: ..
      dockerfile: docker/Dockerfile
    environment:
      PGSQL_DATABASE: ${PGSQL_DATABASE}
      PGSQL_USERNAME: ${PGSQL_USERNAME}
      PGSQL_PASSWORD: ${PGSQL_PASSWORD}
    env_file: .env
    depends_on:
      postgres:
        condition: service_healthy
    ports:
      - "${APP_PORT}:8080"
    networks:
      - EVALIA

volumes:
  evalia_database:

networks:
  EVALIA:
    driver: bridge
